crates/sql-sink/src/main.rs

```diff
@@ -27,49 +29,75 @@ const BACKOFF_MAX: Duration = Duration::from_secs(3600 * 24);
 
 #[connector(sink)]
 async fn start(config: SqlConfig, mut stream: impl ConsumerStream) -> Result<()> {
-    let mut sink = start_sink(config).await?;
-    while let Some(item) = stream.next().await {
-        let operation: Operation = serde_json::from_slice(item?.as_ref())?;
-        trace!(?operation);
-        sink.send(operation).await?;
+    let mut backoff = backoff_init()?;
+    let mut sink = start_sink(&mut backoff, &config).await?;
+
+    info!("Starting to process records");
+
+    while let Some(item_result) = stream.next().await {
+        match item_result {
+            Ok(item) => match serde_json::from_slice::<Operation>(item.as_ref()) {
+                Ok(operation) => {
+                    trace!(?operation);
+                    while let Err(e) = sink.send(operation.clone()).await {
```

Comment:
This should not be while loop. If backoff failed then it should exit

Reply:
Really? Only one retry on db?

Reply:
start_sink has a backoff_and_wait inside, so if keep failing it will exit.